generate_workspace/src/main/java/com/google/devtools/build/workspace/maven/DefaultModelResolver.java

```diff
@@ -187,19 +195,14 @@ public void addRepository(Repository repository) {
 
   @Override
   public void addRepository(Repository repository, boolean replace) {
-    if (repositories.contains(repository)) {
-      if (replace) {
-        repositories.remove(repository);
-        repositories.add(repository);
-      }
-    } else {
-      repositories.add(repository);
+    if (replace || !repositoriesById.containsKey(repository.getId())) {
```

Comment:
nit: can you add a simple comment for the boolean logic?
Although the new version is more concise, it took me a
second to remember the conditions.